### PR TITLE
fix: improve query params sanitization

### PR DESCRIPTION
**Pull request by amateima**
This PR updates validation of the query params in order to restrict `limit` and `offset` within a certain range

### CODE REVIEW

File: src/modules/deposit/dto.ts [hunk 1]
```diff
@@ -16,13 +17,19 @@ export class GetDepositsQuery {
   status: "filled" | "pending";
 
   @IsOptional()
-  @IsNumberString({ no_symbols: true })
-  @ApiProperty({ example: "10", required: false })
+  @IsInt()
+  @Min(1)
+  @Max(100)
```
Comment (amateima) [inline src/modules/deposit/dto.ts:22]: Restrict `?limit` to max 100. The FE client doesn't request more than 50 deposits per table page

File: src/modules/referral/entry-points/http/dto.ts [hunk 2]
```diff
@@ -14,12 +15,18 @@ export class GetReferralsQuery {
   @ApiProperty({ example: "0x9A8f92a830A5cB89a3816e3D267CB7791c16b04D", minLength: 42, maxLength: 42, required: true })
   address: string;
 
-  @IsNumberString()
-  @ApiProperty({ example: "10", required: false })
+  @IsInt()
+  @Min(1)
+  @Max(100)
```
Comment (amateima) [inline src/modules/referral/entry-points/http/dto.ts:20]: Restrict `?limit` to max 100. The FE client doesn't request more than 50 deposits per table page

**Review by dohaki — APPROVED** [review]
lgtm

**Review by pxrl — APPROVED** [review]

File: src/modules/deposit/dto.ts [hunk 3]
```diff
@@ -16,13 +17,19 @@ export class GetDepositsQuery {
...
+  @Type(() => Number)
+  @ApiProperty({ example: 10, required: false })
   limit: string;
 
   @IsOptional()
-  @IsNumberString({ no_symbols: true })
-  @ApiProperty({ example: "0", required: false })
+  @IsInt()
+  @Min(0)
+  @Max(10_000_000)
```
Comment (mrice32) [inline src/modules/deposit/dto.ts:30]: Does a large offset cause any problems? Would 10MM be significantly more expensive than 0?

**Review by mrice32 — COMMENTED** [review]
Looks great! One comment

File: src/modules/deposit/dto.ts (hunk 3 above)
Comment (amateima) [reply src/modules/deposit/dto.ts:30]: It impacts the time execution of the query, yes

**Pull request merged**